query.go

```diff
@@ -300,7 +299,13 @@ func hangingPackages(removeOptional bool) (hanging []string, err error) {
 		}
 
 		pkg.Provides().ForEach(func(dep alpm.Depend) error {
-			provides[dep.Name] = pkg.Name()
```

Comment:
There's a function called `addMapStringSet` that does all this for you ;)

feel free to create `addMapStringString` as well or find a generic way.

Reply:
I was not aware of `addMapStringSet` and I'll rewrite my code.

I'm not too sure what you mean by `addMapStringString`, or which part of my code you think I should use it in.

Reply:
Yeah I wasn't thinking when I said that. Was thinking about `map[stiring]string` but obviously you can set that directly.